src/Components/Concerns/StepAware.php

```diff
@@ -33,5 +34,19 @@ public function bootedStepAware()
                 return new Step($stepName, $info, $status);
             })
             ->toArray();
+
+        $this->getProgress();
+    }
+
+    public function getProgress(): float
```

Comment:
Instead of calculating the percentage of the progress, I would just let the component be aware of the stepnumber and total number of steps.

Formatting this info as a percentage is then a task for the user of the package, and not the package itself.